Firmware/menu.cpp

```diff
@@ -551,4 +553,33 @@ uint8_t menu_item_edit_P(const char* str, T pval, int16_t min_val, int16_t max_v
 template uint8_t menu_item_edit_P<int16_t*>(const char* str, int16_t *pval, int16_t min_val, int16_t max_val);
 template uint8_t menu_item_edit_P<uint8_t*>(const char* str, uint8_t *pval, int16_t min_val, int16_t max_val);
 
-#undef _menu_data
+static uint8_t progressbar_block_count = 0;
```

Comment:
Actually this solution is not that bad - I can't imagine what situation may require having 2 independent progress bars on the LCD. The only syntactic sugar (+a bit of safety of use) may be wrapping the progress bar into a class (ideally a singleton, but gcc 4.9 on AVR creates default 16B locks around singletons, which is not something we want).